### DISCUSSION TITLE
Changing IP address for feeds

### DISCUSSION THREAD

**Discussion by noire-x** [immauss/openvas#257] [Q&A]
Hi there Scott,
i would like to change the IP address from where feeds are being downloaded. I dont want to expose the 873 plain over the internet so i'd rather have a VM on a separate network where i download the feeds and then connect to it via VPN and have my feeds downloaded/upgraded.
Do you know how to do this?
Ive found this:
`echo "45.135.106.142 feed.community.greenbone.net feed.openvas.org" >> /etc/hosts` so therefore i will just need to change with my own IP address.
Thanks!

**Comment by immauss** [comment]
You could try looking here:
https://github.com/greenbone/greenbone-feed-sync?tab=readme-ov-file#feed-url

Though I've never actually tried this myself.  Looks like you would need to set the environment variable or CLI option for each one though. (tedious) Your method might work as well though and be a bit easier, but the info at that URL could help you get it setup correctly. 

I'm not sure what you mean by "expose the 873" though. Running the greenbone-feed-sync should not expose anything and should not require exposing anything either.  Can you explain more?  

-Scott

**Comment by noire-x** [comment]
Hey Scott sure,
rsync protocol runs on the TCP port 873 which is then used by GVM to download feeds. It has to be forwarded/NATted on firewall to your local machine where GVM is installed along with whitelisting the source IP 45.135.106.143.
So i want to avoid all this above having a VM on a separate network where i download the feeds and then connect to it via VPN and have my feeds downloaded/upgraded. :) 
Actually it could work even if there are more of your GVM docker instances on different machines in different networks all linked together via VPN. So you have one master VM where to download the feeds and then all the slaves getting feeds from the master

**Reply by immauss** [reply]
yes .... but .... 
Unless you have a very unusual firewall setup, an outbound TCP connection to port 873 of a server on the internet outside your firewall should not require a change to that firewall. Only in a case where your firewall is limiting the outbound protocols, which is usually only in heavily controlled corporate or government environment.  However, even in those cases, It would not require you "expose" 873 on your own system.
